subs/pantry/src/Pantry/Types.hs

```diff
@@ -205,8 +205,8 @@ newtype Revision = Revision Word
 -- whether a pool is used, and the default implementation in
 -- "Pantry.Storage" does not use a pool.
 data Storage = Storage
-  { withStorage_ :: (forall env a. HasLogFunc env => ReaderT SqlBackend (RIO env) a -> RIO env a)
-  , withWriteLock_ :: (forall env a. HasLogFunc env => RIO env a -> RIO env a)
+  { withStorage_ :: forall m a. MonadUnliftIO m => ReaderT SqlBackend m a -> m a
```

Comment:
I think overall I'd rather push the code in stackage-server to use `RIO` rather than have to avoid its usage here. But I'm open to hearing that that would be too big a change.

Reply:
I like that idea, but I wanna push back on it until we can use RIO with Yesod easily.

Reply:
Ok, I switched it back to the original signature. 
```haskell
withStorage_ :: (forall env a. HasLogFunc env => ReaderT SqlBackend (RIO env) a -> RIO env a)
```
Also adjusted stackage server to use RIO for connecting to db as well.